trio/_core/_multierror.py

```diff
@@ -355,60 +360,72 @@ def format_exception(etype, value, tb, *, limit=None, chain=True):
     thread at any time.
 
     """
-    return _format_exception_multi(set(), etype, value, tb, limit, chain)
+    return traceback.format_exception(
+        etype, value, tb, limit=limit, chain=chain
+    )
 
 
-def _format_exception_multi(seen, etype, value, tb, limit, chain):
-    if id(value) in seen:
-        return ["<duplicate exception {!r}>\n".format(value)]
-    seen.add(id(value))
+traceback_exception_original_init = traceback.TracebackException.__init__
+
+
+def traceback_exception_init(
+        self,
+        exc_type,
+        exc_value,
+        exc_traceback,
+        *,
+        limit=None,
+        lookup_lines=True,
+        capture_locals=False,
+        _seen=None
+):
+    traceback_exception_original_init(
+        self,
+        exc_type,
+        exc_value,
+        exc_traceback,
+        limit=limit,
+        lookup_lines=lookup_lines,
+        capture_locals=capture_locals,
+        _seen=_seen
+    )
 
-    chunks = []
-    if chain:
-        if value.__cause__ is not None:
-            v = value.__cause__
-            chunks += _format_exception_multi(
-                seen, type(v), v, v.__traceback__, limit=limit, chain=True
-            )
-            chunks += [
-                "\nThe above exception was the direct cause of the "
-                "following exception:\n\n",
-            ]
-        elif value.__context__ is not None and not value.__suppress_context__:
-            v = value.__context__
-            chunks += _format_exception_multi(
-                seen, type(v), v, v.__traceback__, limit=limit, chain=True
+    if isinstance(exc_value, MultiError):
+        embedded = []
```

Comment:
There is no explicit `seen` behavior here anymore to log duplicate exceptions. The behavior of `TracebackException` is just to omit it if it has been seen.

Is that okay?